src/pages/index.js

```diff
@@ -21,91 +34,155 @@ popUpImageModal.setEventListeners();
 function handleCardClick(name, link) {
   popUpImageModal.openModal(name, link);
 }
-
+//Popup for deleting cards from API along with handler
+const dltPopup = new PopupWithForm(constants.selectors.deleteSelector);
+dltPopup.setEventListeners();
+function handleDelete(id, card) {
+  dltPopup.openModal();
+  dltPopup.setSubmitAction(() => {
+    api
+      .deleteCards(id)
+      .then(() => {
+        dltPopup.closeModal();
+        card.deleteUI();
+      })
+      .finally(() => {
+        constants.dltFormSubmit.innerText = "Yes";
+      });
+  });
+}
+//Handler for the Like button
+function handleToggleLikes(id, isLiked, card) {
+  if (isLiked === false) {
+    card.handleLike();
+    isLiked = true;
```

Comment:
```js

    card.handleLike();
    isLiked = true;
```
You need to **toggle likes** only in block `then` after a successful response from the server because if there is an error the user will not know about that and will see the toggled heart color.